### PR TITLE
CloudFront Signers Add Snippets and Tests

### PR DESCRIPTION
**Pull request by davidh44**
Adding code snippets and tests for CloudFrontUtilities

## Motivation and Context

Snippets were missing, code coverage was low

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java [hunk 1]
```diff
@@ -75,9 +75,27 @@ public static CloudFrontUtilities create() {
...
+     *            A {@link Consumer} that will call methods on {@link CannedSignerRequest.Builder} to create a request.
      * @return A signed URL that will permit access to a specific distribution
      *         and S3 object.
+     *
+     * <p><b>Example Usage</b>
+     * <p>
+     * {@snippet :
+     *     CloudFrontUtilities utilities = CloudFrontUtilities.create();
+     *
+     *     Instant expirationDate = Instant.now().plus(7, ChronoUnit.DAYS);
```
Comment (dagnir) [inline services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java:88]: `plus(Duration.ofDays(7))`

Let's add a small blurb explaining what the example is illustrating. Ditto for the other examples

File: services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java [hunk 2]
```diff
@@ -95,6 +113,26 @@ public SignedUrl getSignedUrlWithCannedPolicy(Consumer<CannedSignerRequest.Build
      *            resourceUrl, privateKey, keyPairId, expirationDate
      * @return A signed URL that will permit access to a specific distribution
      *         and S3 object.
+     *
+     * <p><b>Example Usage</b>
+     * <p>
+     * {@snippet :
+     *     CloudFrontUtilities utilities = CloudFrontUtilities.create();
+     *
+     *     Instant expirationDate = Instant.now().plus(7, ChronoUnit.DAYS);
```
Comment (dagnir) [inline services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java:122]: Duration.ofDays

File: services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java (hunk 1 above)
Comment (davidh44) [reply services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java]: Added small descriptions

File: services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java [hunk 3]
```diff
@@ -75,9 +75,27 @@ public static CloudFrontUtilities create() {
...
+     *     String resourceUrl = "https://d111111abcdef8.cloudfront.net/s3ObjectKey";
+     *     String keyPairId = "myKeyPairId";
+     *     PrivateKey privateKey = myPrivateKey; // Either PrivateKey or Path can be passed in
```
[the remark this answers is not in the document]
Comment (davidh44) [reply services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java]: removed, documented already

File: services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java [hunk 4]
```diff
@@ -130,9 +168,31 @@ public SignedUrl getSignedUrlWithCannedPolicy(CannedSignerRequest request) {
...
+     *     Instant activeDate = Instant.now().plus(2, ChronoUnit.DAYS);
+     *     String ipRange = "192.168.0.1/24";
+     *
+     *     SignedUrl signedUrl = utilities.getSignedUrlWithCustomPolicy(r -> r.resourceUrl(resourceUrl)
+     *                                                                        .privateKey(privateKey)
+     *                                                                        .keyPairId(keyPairId)
+     *                                                                        .expirationDate(expirationDate)
+     *                                                                        .activeDate(activeDate) //optional
```
[the remark this answers is not in the document]
Comment (davidh44) [reply services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java]: removed, documented already

File: services/cloudfront/src/test/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilitiesTest.java [hunk 5]
```diff
@@ -108,4 +160,61 @@ void getSignedURLWithCustomPolicy_shouldWork() throws Exception {
...
+            cloudFrontUtilities.getSignedUrlWithCustomPolicy(r -> r
+                .resourceUrl(resourceUrl)
+                .privateKey(keyPair.getPrivate())
+                .keyPairId("keyPairId"))
+        );
+        assertThat(exception.getMessage().contains("Expiration date must be provided to sign CloudFront URLs"));
+    }
+
+    @Test
+    void getCookiesForCannedPolicy_shouldWork() throws Exception {
```
[the remark this answers is not in the document]
Comment (davidh44) [reply services/cloudfront/src/test/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilitiesTest.java]: updated to more specific test names

File: services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java (hunk 2 above)
Comment (davidh44) [reply services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java]: updated

**Review by dagnir — APPROVED** [review]

**Pull request merged**